packages/office-ui-fabric-react/src/components/Shimmer/Shimmer.types.ts

```diff
@@ -0,0 +1,213 @@
+import * as React from 'react';
+import { IStyle, ITheme } from '../../Styling';
+import { IStyleFunctionOrObject } from '../../Utilities';
+
+export interface IShimmer {}
+
+/**
+ * Shimmer component props.
+ */
+export interface IShimmerProps extends React.AllHTMLAttributes<HTMLElement> {
+  /**
+   * Optional callback to access the IShimmer interface. Use this instead of ref for accessing
+   * the public methods and properties of the component.
+   */
+  componentRef?: (component: IShimmer | null) => void;
+
+  /**
+   * Sets the width of the shimmer wave wrapper in percentages relative to the containig parent element.
+   * @default 100%
+   */
+  widthInPercentage?: number;
+
+  /**
+   * Sets the width of the shimmer wave wrapper to an exact value in pixels.
+   * If none of the widths provided, it defaults to 100%.
+   */
+  widthInPixel?: number;
+
+  /**
+   * Controls when the shimmer is swapped with actual data through an animated transition.
+   * @default false
+   */
+  isDataLoaded?: boolean;
+
+  /**
+   * Elements to render in one line of the Shimmer.
+   */
+  shimmerElements?: IShimmerElement[];
+
+  /**
+   * Custom elements when necessary to build complex placeholder skeletons.
+   */
+  customElementsGroup?: React.ReactNode;
```

Comment:
Where is this placed within the overall hierarchy of the shimmer? Which elements does this affect?

Reply:
@JasonGore This prop is almost doing the same thing as `shimmerElements` prop with the only difference being semantics, so that whenever the user is building a more complex (custom) skeleton using the `ShimmerElementsGroup` subcomponent multiple times. I have an example shown in `Shimmer.CustomElements.Example.tsx`. For basic scenarios when it's just one line with a couple of elements user should use `shimmerElements` prop for a quick an easy use of Shimmer component, which internally uses just once `ShimmerElementsGroup`. I hope I answered your question... if not I would be happy to elaborate more.